### PR TITLE
add message types

### PR DESCRIPTION
**Pull request by teaforthecat**
Add collection and instance resources for the new `message_types` endpoint

### CODE REVIEW

File: lib/govdelivery/tms/resource/message_type.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module GovDelivery::TMS #:nodoc:
+  # A MessageType is a like a tag that can be attached to an EmailTemplate or an
+  # EmailMessage. It is included in encoded links and can be used to as a
```
Comment (unclebilly) [inline lib/govdelivery/tms/resource/message_type.rb]: There is a grammar problem here...

**Comment by teaforthecat** [comment]
after some tweaks, this is ready

File: lib/govdelivery/tms/resource/email_message.rb [hunk 2]
```diff
@@ -50,6 +50,8 @@ class EmailMessage
                          :from_email,
                          :from_name,
                          :macros,
+                         :message_type_code,
+                         :message_type_label,
```
Comment (zeckalpha) [inline lib/govdelivery/tms/resource/email_message.rb]: This should be a readonly_attribute.

File: lib/govdelivery/tms/resource/email_template.rb [hunk 3]
```diff
@@ -3,9 +3,17 @@ class EmailTemplate
     include InstanceResource
 
     # @!parse attr_accessor :body, :subject, :link_tracking_parameters, :macros, :open_tracking_enabled, :click_tracking_enabled
-    writeable_attributes :body, :subject, :link_tracking_parameters, :macros, :open_tracking_enabled, :click_tracking_enabled, :uuid
+    writeable_attributes :body,
+                         :subject,
+                         :link_tracking_parameters,
+                         :macros,
+                         :message_type_code,
+                         :message_type_label,
```
Comment (zeckalpha) [inline lib/govdelivery/tms/resource/email_template.rb]: Same here.

**Pull request merged**